lib/iris/fileformats/grib/__init__.py

```diff
@@ -928,29 +927,36 @@ def save_grib2(cube, target, append=False, **kwargs):
 
     Args:
 
-        * cube      - A :class:`iris.cube.Cube`, :class:`iris.cube.CubeList` or list of cubes.
+        * cube      - A :class:`iris.cube.Cube`, :class:`iris.cube.CubeList` or
+                      list of cubes.
         * target    - A filename or open file handle.
 
     Kwargs:
 
-        * append    - Whether to start a new file afresh or add the cube(s) to the end of the file.
-                      Only applicable when target is a filename, not a file handle.
-                      Default is False.
+        * append    - Whether to start a new file afresh or add the cube(s) to
+                      the end of the file.
+                      Only applicable when target is a filename, not a file
+                      handle. Default is False.
 
     See also :func:`iris.io.save`.
 
     """
+    messages = as_messages(cube)
+    save_messages(messages, target, append=append)
 
-    # grib file (this bit is common to the pp and grib savers...)
-    if isinstance(target, basestring):
-        grib_file = open(target, "ab" if append else "wb")
-    elif hasattr(target, "write"):
-        if hasattr(target, "mode") and "b" not in target.mode:
-            raise ValueError("Target not binary")
-        grib_file = target
-    else:
-        raise ValueError("Can only save grib to filename or writable")
 
+def as_pairs(cube):
+    """
+    Convert one or more cubes to (2D cube, GRIB message) pairs.
+    Returns an iterable of tuples each consisting of one 2D cubes and
+    one GRIB message, the result of the 2D cube being processed by the GRIB
```

Comment:
I'm uneasy about using the term "GRIB message" to describe what is returned/consumed in these interfaces. They aren't really messages at all, just IDs. How about "GRIB message ID" or "grib_api message ID"?